### PR TITLE
fix test_auto_create_auto_join_where_no_consent

### PR DESCRIPTION
**Pull request by neilisfragile**

### CODE REVIEW

File: tests/handlers/test_register.py [hunk 1]
```diff
@@ -187,18 +187,30 @@ def test_auto_create_auto_join_rooms_when_support_user_exists(self):
...
     def test_auto_create_auto_join_where_no_consent(self):
-        """XXX what is this trying to test? I *think* it is trying to test
-        that we are not auto-joined to the server notices room if
-        block_events_without_consent_error is set, but (a) that doesn't seem to be
-        a sensible thing to test for, and (b) it doesn't work anyway because you can't
-        change the config after the EventCreationHandler has been instantiated.
+        """Test to ensure that the first user is not auto-joined to a room if
+        they have not given general consent.
         """
         self.hs.config.user_consent_at_registration = True
```
Comment (richvdh) [inline tests/handlers/test_register.py]: does this and the next line actually do anything? I'd expect them to be redundant now.

File: tests/handlers/test_register.py [hunk 2]
```diff
@@ -187,18 +187,30 @@ def test_auto_create_auto_join_rooms_when_support_user_exists(self):
...
         self.hs.config.block_events_without_consent_error = "Error"
+
+        # Given:-
+        #    * a user must give consent,
+        #    * they have not given that consent
+        #    * The server is configured to auto-join to a room
+        # (and autocreate if necessary)
+        event_creation_handler = self.hs.get_event_creation_handler()
+        event_creation_handler._block_events_without_consent_error = (
```
Comment (richvdh) [inline tests/handlers/test_register.py]: I'm a bit dubious about this poking into the guts of `EventCreationHandler` - it looks fragile. OTOH I can't really think of a better way of testing it, other than creating a new test class (possibly a subclass of `RegistrationTestCase`) which uses a different config - which might be a bit fiddly right now.

File: tests/handlers/test_register.py [hunk 3]
```diff
@@ -187,18 +187,30 @@ def test_auto_create_auto_join_rooms_when_support_user_exists(self):
...
+            self.hs.config.block_events_without_consent_error
+        )
+        event_creation_handler._consent_uri_builder = Mock()
         room_alias_str = "#room:test"
         self.hs.config.auto_join_rooms = [room_alias_str]
+
+        # When the user is registered, and post consent actions are called
```
Comment (richvdh) [inline tests/handlers/test_register.py]: this doesn't seem to make sense?

File: tests/handlers/test_register.py (hunk 1 above)
Comment (neilisfragile) [reply tests/handlers/test_register.py]: Nope

File: tests/handlers/test_register.py (hunk 2 above)
Comment (neilisfragile) [reply tests/handlers/test_register.py]: Yeah this really mings - I'll comment explicitly to make it clearer

File: tests/handlers/test_register.py (hunk 3 above)
Comment (neilisfragile) [reply tests/handlers/test_register.py]: I've tried to make it clearer with formatting

**Comment by richvdh** [comment]
(lgtm other than the changelog)

**Pull request merged**